kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableSchemaEvolutionSuite.scala

```diff
@@ -1160,33 +1200,68 @@ class DeltaTableSchemaEvolutionSuite extends DeltaTableWriteSuiteBase with Colum
     }
   }
 
-  // TODO: need to remove this once we support schema evolution with clustering columns
-  test("Cannot update schema with clustering columns") {
-    withTempDirAndEngine { (tablePath, engine) =>
-      val table = Table.forPath(engine, tablePath)
-      val initialSchema = new StructType()
-        .add("a", StringType.STRING, true)
-        .add("c", IntegerType.INTEGER, true)
-
-      createEmptyTable(
-        engine,
-        tablePath,
-        initialSchema,
-        clusteringCols = List(new Column("c")),
-        tableProperties = Map(
-          TableConfig.COLUMN_MAPPING_MODE.getKey -> "id",
-          TableConfig.ICEBERG_COMPAT_V2_ENABLED.getKey -> "true"))
-
-      val currentSchema = table.getLatestSnapshot(engine).getSchema
-      val newSchema = new StructType()
-        .add("c", StringType.STRING, true, currentSchema.get("c").getMetadata)
-        .add("a", StringType.STRING, true, currentSchema.get("a").getMetadata)
-
-      assertSchemaEvolutionFails[KernelException](
-        table,
-        engine,
-        newSchema,
-        "Update schema for table with clustering columns")
+  val primitiveSchemaWithClusteringColumn = new StructType()
+    .add("clustering_col", IntegerType.INTEGER, fieldMetadataForColumn(1, "clustering_col"))
+    .add("data", IntegerType.INTEGER, fieldMetadataForColumn(2, "data"))
+
+  val nestedSchemaWithClusteringColumn = new StructType()
+    .add(
+      "struct",
+      new StructType()
+        .add("clustering_col", IntegerType.INTEGER, fieldMetadataForColumn(1, "clustering_col"))
+        .add("data", IntegerType.INTEGER, fieldMetadataForColumn(2, "data")),
+      true,
+      fieldMetadataForColumn(3, "struct"))
+
+  private val updatedSchemaWithDroppedClusteringColumn = Tables.Table(
+    ("schemaBefore", "updatedSchemaWithDroppedClusteringColumn", "clusteringColumn"),
+    (
+      primitiveSchemaWithClusteringColumn,
+      new StructType()
+        .add(
+          "data",
+          IntegerType.INTEGER,
+          true,
+          primitiveSchemaWithClusteringColumn.get("data").getMetadata),
+      new Column("clustering_col")),
+    (
+      nestedSchemaWithClusteringColumn,
+      new StructType()
```

Comment:
not necessary for this PR but it might simplify the tests a lot (not just here, in all the schema code) if we added a test utils that supported something like `schema.withColumnDropped(x)` -- then we don't need to redefine the schema in each place and can just drop columns!

Reply:
just an idea to improve the tests -- maybe note this down since I think you're already tracking test improvements

Reply:
Yeah I was planning on doing this on the test cleanup PR, I think the util in test to just drop/rename by field name is a fine way of handling things for now.

 I think in the long run, with a visitor we'd be able to more generically apply a set of whatever changes to the schema and produce a new schema (and this would also be on the general schema update path as well) so we don't have to redefine things.